src/gurt/misc.c

```diff
@@ -35,31 +61,72 @@ d_rand()
 void
 d_free(void *ptr)
 {
+	if (unlikely(d_free_track_cb != NULL && track_arg != NULL)) {
+		size_t size = malloc_usable_size(ptr);
+
+		d_free_track_cb(track_arg, size);
+	}
+
 	free(ptr);
 }
 
 void *
 d_calloc(size_t count, size_t eltsize)
 {
-	return calloc(count, eltsize);
+	void *ptr;
+
+	ptr = calloc(count, eltsize);
+	if (unlikely(d_alloc_track_cb && track_arg != NULL)) {
+		if (ptr != NULL)
+			d_alloc_track_cb(track_arg, eltsize);
+	}
```

Comment:
You could probably get away with only checking track_arg and not d_alloc_track_cb here as well - but you might have to update d_set_alloc_track_arg to check if the callbacks have been set.
```suggestion
	if (unlikely(d_alloc_track_cb && track_arg != NULL && prt != NULL))
		d_alloc_track_cb(track_arg, eltsize);
```